### PR TITLE
JAMES-3955 Check for queues without consumer and resume consumption a…

### PR DESCRIPTION
**Pull request by chibenwa**
…s needed

### CODE REVIEW

**Review by Arsnael — APPROVED** [review]

File: event-bus/distributed/src/main/java/org/apache/james/events/RabbitEventBusConsumerHealthCheck.java [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+
+        Optional<String> queueWithoutConsumers = groups
+            .map(namingStrategy::workQueue)
+            .map(GroupRegistration.WorkQueueName::asString)
+            .filter(Throwing.predicate(queue -> channel.consumerCount(queue) == 0))
+            .findAny();
+
+        if (queueWithoutConsumers.isPresent()) {
+            eventBus.restart();
+            return Result.degraded(COMPONENT, "No consumers on " + queueWithoutConsumers.get());
```
Comment (quantranhong1999) [inline event-bus/distributed/src/main/java/org/apache/james/events/RabbitEventBusConsumerHealthCheck.java:80]: Should we add a log upon reconnecting the consumer?

File: server/queue/queue-rabbitmq/src/main/java/org/apache/james/queue/rabbitmq/RabbitMQMailQueueConsumerHealthCheck.java [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+            .map(m -> m.toWorkQueueName().asString())
+            .anyMatch(Throwing.predicate(queue -> channel.consumerCount(queue) == 0));
+
+        if (queueWithoutConsumers) {
+            reconnectionHandlers.forEach(r -> connectionPool.getResilientConnection()
+                .flatMap(c -> Mono.from(r.handleReconnection(c)))
+                .subscribeOn(Schedulers.boundedElastic())
+                .block());
+
+            return Result.degraded(COMPONENT, "No consumers");
```
Comment (quantranhong1999) [inline server/queue/queue-rabbitmq/src/main/java/org/apache/james/queue/rabbitmq/RabbitMQMailQueueConsumerHealthCheck.java:82]: Should we add a log upon reconnecting?

**Review by quantranhong1999 — APPROVED** [review]

**Comment by Arsnael** [comment]
https://ci-builds.apache.org/job/james/job/ApacheJames/job/PR-1778/5/testReport/junit/org.apache.james.webadmin.integration.rabbitmq/RabbitMQWebAdminServerIntegrationImmutableTest/healthCheckComponentsList/

Related :)

File: server/queue/queue-rabbitmq/src/main/java/org/apache/james/queue/rabbitmq/RabbitMQMailQueueConsumerHealthCheck.java (hunk 2 above)
Comment (chibenwa) [reply server/queue/queue-rabbitmq/src/main/java/org/apache/james/queue/rabbitmq/RabbitMQMailQueueConsumerHealthCheck.java:82]: Periodical health check already emits a log

**Review by Arsnael — APPROVED** [review]

**Pull request merged**